configure.ac

```diff
@@ -1671,6 +1671,37 @@ if test "x${je_cv_gcc_builtin_popcountl}" = "xyes" ; then
   AC_DEFINE([JEMALLOC_INTERNAL_POPCOUNTLL], [__builtin_popcountll], [ ])
 fi
 
+dnl Configure maximum number of tcaches/arenas
+AC_ARG_WITH([lg_tcache_limit],
+  [AS_HELP_STRING([--with-lg-tcache-limit=<lg-cache-limit>],
+  [Base 2 log of maximum number of tcaches (between 12 and 15). Trade off with number of arenas, which will be 2 ^ (24 - lg-cache-limit).])])
+
+if test "x$with_lg_tcache_limit" = "x"; then
+  # Default is 4096 tcaches, 4096 arenas
+  with_lg_tcache_limit=12
+fi
+
+if ! test "$with_lg_tcache_limit" -ge 12 -a "$with_lg_tcache_limit" -le 15; then
```

Comment:
This feels over-specified to me. If we are giving the user (of jemalloc) the choice of how many tcaches / arenas they want, why stop them having more than 15 (or less than 12?)

Suggest allowing it to be configured up to the limit of what jemalloc fundamentally supports - i.e. maybe at least 1 bit for arena and at least one for tcache?

Reply:
Having less than some number of arenas (17 with my unit test changes) makes some of the unit tests fail.